### PR TITLE
Add node driver affinity and topology support

### PR DESCRIPTION
**Pull request by guangbochen**
Related Issues:
- https://github.com/harvester/harvester/issues/2316
- https://github.com/harvester/harvester/issues/1418

### CODE REVIEW

**Review by divya-mohan0209 — COMMENTED** [review]
Few suggestions, I hope you find them helpful!

**Comment by guangbochen** [comment]
Updated, thanks a lot for the review.

File: docs/rancher/k3s-cluster.md [hunk 1]
```diff
@@ -38,6 +38,31 @@ You can create a K3s Kubernetes cluster from the **Cluster Management** page via
...
+_Available as of v1.0.3 + Rancher v2.6.7_
+
+The Harvester node driver now supports scheduling a group of machines to particular nodes through the node affinity rules. This provides high availability and better resource utilization.
+
+Node affinity can be added to the machine pools during the cluster creation:
+
+1. Click the `Show Advanced` button and click the `Add Node Selector`
+   ![affinity-add-node-selector](assets/affinity-rke2-add-node-selector.png)
+2. Set priority to `Required` if you wish the scheduler to schedule the machines only when the rules are met.
+3. Click `Add Rule` to specify the node affinity rules, e.g., for the [topology spread constraints](./node-driver.md/#node-driver-topology-spread-constraints) use case, you can add the `region` and `zone` labels as follows:
```
Comment (chrisho) [inline docs/rancher/k3s-cluster.md]: Quick link is wrong, should be `./node-driver.md/#topology-spread-constraints`

**Review by chrisho — APPROVED** [review]
LGTM

**Review by divya-mohan0209 — APPROVED** [review]
lgtm!

**Pull request merged**